controllers/topology/reconcile_state_test.go

```diff
@@ -678,8 +682,8 @@ func TestReconcileMachineDeployments(t *testing.T) {
 		{
 			name:    "Should update MachineDeployment with InfrastructureMachineTemplate and BootstrapTemplate without rotation",
 			current: []*scope.MachineDeploymentState{md4m},
-			desired: []*scope.MachineDeploymentState{md4mWithRotatedTemplates},
-			want:    []*scope.MachineDeploymentState{md4m},
+			desired: []*scope.MachineDeploymentState{md4mWithInPlaceUpdatedTemplates},
+			want:    []*scope.MachineDeploymentState{md4mWithInPlaceUpdatedTemplates},
```

Comment:
This test was wrong, but the side effect in mergepatch.Patch was hiding the error